presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -145,6 +147,34 @@ protected PlanNodeStatsEstimate visitBooleanLiteral(BooleanLiteral node, Void co
             }
         }
 
+        @Override
+        protected PlanNodeStatsEstimate visitIsNotNullPredicate(IsNotNullPredicate node, Void context)
+        {
+            if (node.getValue() instanceof SymbolReference) {
+                Symbol symbol = Symbol.from(node.getValue());
+                SymbolStatsEstimate symbolStatsEstimate = input.getSymbolStatistics(symbol);
+                return input.mapOutputRowCount(rowCount -> rowCount * (1 - symbolStatsEstimate.getNullsFraction()))
+                        .mapSymbolColumnStatistics(symbol, statsEstimate -> statsEstimate.mapNullsFraction(x -> 0.0));
+            }
+            return visitExpression(node, context);
```

Comment:
IMO returning here unknown stats directly is more readable.

Reply:
I agree. Later.